data/abilities.js

```diff
@@ -259,7 +257,8 @@ let BattleAbilities = {
 		desc: "If this Pokemon is a Greninja, it transforms into Ash-Greninja after knocking out a Pokemon. As Ash-Greninja, its Water Shuriken has 20 base power and always hits 3 times.",
 		shortDesc: "After KOing a Pokemon: becomes Ash-Greninja, Water Shuriken: 20 power, hits 3x.",
 		onSourceFaint(target, source, effect) {
-			if (effect && effect.effectType === 'Move' && source.template.speciesid === 'greninja' && source.hp && !source.transformed && source.side.foe.pokemonLeft) {
+			const remainingFoes = this.sides.reduce((total, side) => total + (+(side.n % 2 !== source.side.n % 2) && side.pokemonLeft), 0);
```

Comment:
I dislike all reductions that have an accumulator, and this one has an accumulator. I think accumulators are much more readable as `for`-loops.

Reply:
Also we should maybe support `side.foes()` and `side.allies()`?

Reply:
Also in theory this should delegate to whatever checks for the end of the battle.

Reply:
Yes, it should, but that seems like something for another PR, no?

Reply:
I rather like reducers like this; you've probably noticed that I tend to use them. Although maybe you haven't noticed since I actually apparently snuck two of them by you already.

Reply:
would `side.foes()` return an array of `side`s?

Reply:
Yes